src/ILCompiler.Compiler/src/Compiler/CompilerTypeSystemContext.cs

```diff
@@ -238,7 +238,7 @@ private EcmaModule AddModule(string filePath, string expectedSimpleName)
                     ModuleData actualModuleData = _simpleNameHashtable.AddOrGetExisting(moduleData);
                     if (actualModuleData != moduleData)
                     {
-                        if (actualModuleData.FilePath != filePath)
+                        if (actualModuleData.FilePath != filePath && actualModuleData.SimpleName != "mscorlib")
```

Comment:
If we're going with a hack, we should:
1. Make sure it's tracked by a GitHub issue and add a link to the GitHub issue to the code.
2. Put the hack as close as possible to the source of the problem. In this case, instead of trying to compensate for bad data being in the system, let's prevent them from entering the system. Can you just filter this right after we parse the input in `Program.cs`?